packages/expo-notifications/ios/EXNotifications/Notifications/NotificationCenterManager.swift

```diff
@@ -0,0 +1,131 @@
+import ExpoModulesCore
+import Foundation
+
+/**
+ Protocols that NotificationCenterManager delegates may implement
+ */
+public protocol NotificationPresentationDelegate: AnyObject {
+  func willPresent(_ notification: UNNotification, completionHandler: @escaping (UNNotificationPresentationOptions) -> Void)
+}
+
+public protocol NotificationResponseDelegate: AnyObject {
+  func didReceive(_ response: UNNotificationResponse, completionHandler: @escaping () -> Void)
+}
+public protocol NotificationSettingsDelegate: AnyObject {
+  func openSettings(_ notification: UNNotification?)
+}
+
+public protocol NotificationRegistrationSuccessDelegate: AnyObject {
+  func didRegister(_ deviceToken: String)
+}
+
+public protocol NotificationRegistrationFailureDelegate: AnyObject {
+  func didFailRegistration(_ error: Error)
+}
```

Comment:
Do we need so many protocols? I believe we can have one and provide default (empty) implementations in the protocol extension.

Reply:
I was attempting to duplicate the previous Objective-C code that would see if the methods were implemented -- I agree empty implementations will be simpler